lightly/utils/scheduler.py

```diff
@@ -48,7 +63,7 @@ def cosine_schedule(
             * (np.cos(np.pi * step / (max_steps - 1)) + 1)
             / 2
         )
-    return decay
+    return float(decay)
```

Comment:
Are you sure the conversion to float is required?

Reply:
I had gotten a warning from isort when running the make checks, regarding a potential return path which is not of type float. Honestly I have no idea how there could be one, but I added this type to fix it for now.

Reply:
Ah I see, this is an issue from mypy because `decay` is defined in multiple places using `if/else`. I added a line to define it before the `if/else` which solves the issue.